### ISSUE TITLE
dropout do not take effect

### ISSUE THREAD

**Issue by huangjianaa** [ultralytics/ultralytics#21669]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

To enhance the generalization of different datasets, I attempted to increase the dropout from 0 to 0.1 and 0.5. However, the evaluation results were consistent with those when dropout=0. Why didn't my dropout rate take effect?

### Additional

from ultralytics import YOLO

model = YOLO(str(THIS_DIR/'yolo11x.pt'))  # load a pretrained model (recommended for training)

results = model.train(data=str(THIS_DIR/'datasets/TACO/TACO.yaml'), 
                    epochs=600,
                    imgsz=1960,
                    batch=24,
                    device=[0,1],
                    workers=64,
                    cache=True,
                    project=str(THIS_DIR/'runs/train'),
                    name='exp',
                    optimizer='auto',
                    seed=42,
                    cos_lr=False,
                    amp=True,
                    lr0=0.01,
                    plots=True,
                    weight_decay=0.0005,
                    patience=100,
                    close_mosaic=100,
                    single_cls=True,
                    dropout=0.5,
                    )

task: detect
mode: train
model: /home/gpu08/hj/ultralytics/yolov11/yolo11x.pt
data: /home/gpu08/hj/ultralytics/yolov11/datasets/TACO/TACO.yaml
epochs: 600
time: null
patience: 100
batch: 24
imgsz: 1960
save: true
save_period: -1
cache: true
device: 0,1
workers: 64
project: /home/gpu08/hj/ultralytics/yolov11/runs/train
name: exp46
exist_ok: false
pretrained: true
optimizer: auto
verbose: true
seed: 42
deterministic: true
single_cls: true
rect: false
cos_lr: false
close_mosaic: 100
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.1
val: true
split: val
save_json: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: true
opset: null
workspace: null
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
cutmix: 0.0
copy_paste: 0.0
copy_paste_mode: flip
auto_augment: randaugment
erasing: 0.4
cfg: null
tracker: botsort.yaml
save_dir: /home/gpu08/hj/ultralytics/yolov11/runs/train/exp46

**Comment by UltralyticsAssistant** [comment]
👋 Hello @huangjianaa, thanks for opening this issue and for using Ultralytics 🚀! This is an automated response to help you get support quickly. An Ultralytics engineer will follow up soon with more details.

We recommend a visit to the Docs for general guidance and examples:
- Docs: https://docs.ultralytics.com/
- Python usage: https://docs.ultralytics.com/usage/python/
- CLI usage: https://docs.ultralytics.com/usage/cli/

If this is a 🐛 Bug Report, please provide a minimum reproducible example (MRE) to help us debug it: https://docs.ultralytics.com/help/minimum-reproducible-example/. Include:
- Exact code snippet you ran
- Full console logs with warnings/errors
- Precise ultralytics version and environment details
- A tiny sample dataset or synthetic data reproducing the behavior

If this is a custom training ❓ Question, please share:
- Your dataset details and a few example images
- Full training/validation logs and final metrics
- The exact model/config used and any modified settings
- Confirmation you reviewed our Tips for Best Training Results: https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best:
- Discord for real-time chat: https://discord.com/invite/ultralytics 🎧
- Discourse for in-depth discussions: https://community.ultralytics.com/
- Subreddit for community threads: https://www.reddit.com/r/Ultralytics/

## Upgrade

Please ensure you’re on the latest ultralytics package and requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify if the issue persists on the newest release:

```bash
pip install -U ultralytics
```

Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml

## Environments

YOLO can be run in the following verified environments (with dependencies like CUDA/CUDNN, Python, and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all Ultralytics CI tests are currently passing: https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us investigate the “dropout do not take effect” behavior specifically, please:
- Confirm your ultralytics version and exact command/call used (include the printed args summary)
- Share a minimal script reproducing the issue and a very small dataset or synthetic sample
- Include a short training run log showing the parsed args where dropout is set, plus validation metrics before/after
- Mention any custom model YAML or pretrained weights used (e.g., yolo11x.pt) and any architecture changes

Thanks for the detailed config you’ve shared already 🙏. Once you provide the MRE and environment/version details, we’ll be able to assist more precisely.

**Comment by Y-T-G** [comment]
It's only applicable for classification training:

https://docs.ultralytics.com/modes/train/#train-settings

**Comment by huangjianaa** [comment]
> It's only applicable for classification training:
> 
> https://docs.ultralytics.com/modes/train/#train-settings

How can I open Dropout in detection?

**Comment by Y-T-G** [comment]
You will need to manually add dropout layers to the model, but this will break pretrained weights loading:

https://github.com/ultralytics/ultralytics/issues/15652#issuecomment-2294907995
